config/compiler/alerting.yaml

```diff
@@ -55,3 +55,22 @@ passes:
       comments:
         - 'The amount of time, in seconds, for which the rule must be breached for the rule to be considered to be Firing.'
         - 'Before this time has elapsed, the rule is only considered to be Pending.'
+  - retype_field:
+      field: alerting.TimeInterval.time_intervals
+      as:
+        kind: array
+        array:
+          value_type:
+            kind: ref
+            ref: { referred_pkg: alerting, referred_type: TimeIntervalItem }
+  - retype_object:
+      object: alerting.MatchType
+      as:
+        kind: enum
+        enum:
+          values:
+            - { type: { kind: scalar, scalar: { scalar_kind: string } }, value: '=' }
```

Comment:
Enum members have names :)

https://github.com/grafana/cog/blob/56dd13d47a04682cee1c170c950f6862b4903222/internal/ast/types.go#L651

Reply:
how and where are they used?

Reply:
In jennies, to generate stuff like: https://github.com/grafana/cog/blob/56dd13d47a04682cee1c170c950f6862b4903222/testdata/jennies/rawtypes/enums/GoRawTypes/enums/types_gen.go#L6